edgelet/iotedge/src/system.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::ffi::OsStr;
+
+use lazy_static::lazy_static;
+
+use aziotctl_common::{
+    get_status, get_system_logs as logs, restart, set_log_level as log_level, ServiceDefinition,
+    SERVICE_DEFINITIONS as IS_SERVICES,
+};
+
+use crate::error::{Error, ErrorKind};
+
+lazy_static! {
+    static ref IOTEDGED: ServiceDefinition = {
+        let sockets: &'static [&'static str] = option_env!("IOTEDGE_HOST").map_or(
+            &["aziot-edged.mgmt.socket", "aziot-edged.workload.socket"],
+            |_host| &[],
+        );
+
+        ServiceDefinition {
+            service: "aziot-iotedgd.service",
+            sockets,
+        }
+    };
+    static ref SERVICE_DEFINITIONS: Vec<&'static ServiceDefinition> = {
+        let iotedged: &ServiceDefinition = &IOTEDGED;
+
+        let service_definitions: Vec<&ServiceDefinition> = std::iter::once(iotedged)
+            .chain(IS_SERVICES.iter().copied())
+            .collect();
+
+        service_definitions
+    };
+}
+
+pub struct System;
+
+impl System {
+    pub fn get_system_logs(args: &[&OsStr]) -> Result<(), Error> {
+        let services: Vec<&str> = SERVICE_DEFINITIONS.iter().map(|s| s.service).collect();
+
+        logs(&services, &args).map_err(|err| {
+            eprintln!("{:#?}", err);
+            Error::from(ErrorKind::System)
+        })
+    }
+
+    pub fn system_restart() -> Result<(), Error> {
+        restart(&SERVICE_DEFINITIONS).map_err(|err| {
+            eprintln!("{:#?}", err);
+            Error::from(ErrorKind::System)
+        })
+    }
+
+    pub fn set_log_level(level: log::Level) -> Result<(), Error> {
+        log_level(&SERVICE_DEFINITIONS, level).map_err(|err| {
+            eprintln!("{:#?}", err);
+            Error::from(ErrorKind::System)
+        })
+    }
+
+    pub fn get_system_status() -> Result<(), Error> {
```

Comment:
So what happens on CentOS 7 if `aziot-edged.service` is `inactive` ? Does this succeed even though there are no sockets around to start it up?

Reply:
(If the answer is "yes", don't fix that here by special-casing `IOTEDGED`. Fix it in i-i-s for any `ServiceDefinition` with 0 `socket`s.)

Reply:
yes: https://github.com/Azure/iot-identity-service/pull/119/files